src/node/interfaces.cpp

```diff
@@ -398,6 +399,21 @@ class NodeImpl : public Node
     ArgsManager& args() { return *Assert(Assert(m_context)->args); }
     ChainstateManager& chainman() { return *Assert(m_context->chainman); }
     NodeContext* m_context{nullptr};
+    bool validateProxyAddress(const std::string ipAddress) override
+    {
+        uint16_t port{0};
+        std::string hostname;
+        if (!SplitHostPort(ipAddress, port, hostname) || !port) return false;
+
+        CService serv(LookupNumeric(ipAddress, DEFAULT_PROXY_PORT));
```

Comment:
Ok, it would make sense to remove the call `SplitHostPort()` then, also because we are not returning what the cause of the invalid address is. I'll also add comments on the intention of this logic.